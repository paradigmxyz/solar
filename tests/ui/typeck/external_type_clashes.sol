contract A {
    struct S0 {
        address a;
    }
    struct S1 {
        uint256 a;
        S0 s;
    }
    struct S2 {
        uint256 b;
        S0 s;
    }

    S2 public b;

    function f(S1 memory a) external {}
    //~^ ERROR: function overload clash during conversion to external types for arguments

    function f(S2 memory a) external {}
    //~^ HELP: other declaration is here
}

contract C {
    enum a {
        X
    }

    function f(a) public {}
    //~^ HELP: other declaration is here
}

contract D is C {
    function f(uint8 a) public {}
    //~^ ERROR: function overload clash during conversion to external types for arguments
}

contract E {
    function f(address) external pure {}
    //~^ ERROR: function overload clash during conversion to external types for arguments
    
    function f(address payable) external pure {}
    //~^ HELP: other declaration is here

    function c(address) public pure {}
    //~^ ERROR: function overload clash during conversion to external types for arguments
    
    function c(address payable) public pure {}
    //~^ HELP: other declaration is here
}

type MyAddress is address;

interface I {}

contract F {
    function f(MyAddress a) external {}
    //~^ ERROR: function overload clash during conversion to external types for arguments

    function f(address a) external {}
    //~^ HELP: other declaration is here
}

contract G {
    function g(MyAddress a) external {}
    //~^ HELP: other declaration is here
}

contract H is G {
    function g(I a) external {}
    //~^ ERROR: function overload clash during conversion to external types for arguments
}
