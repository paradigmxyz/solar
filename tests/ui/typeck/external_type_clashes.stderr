error[9914]: function overload clash during conversion to external types for arguments
  --> ROOT/tests/ui/typeck/external_type_clashes.sol:LL:CC
   |
LL |     function f(S1 memory a) external {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
LL |
LL |
LL |     function f(S2 memory a) external {}
   |     ----------------------------------- help: other declaration is here
   |

error[9914]: function overload clash during conversion to external types for arguments
  --> ROOT/tests/ui/typeck/external_type_clashes.sol:LL:CC
   |
LL |     function f(a) public {}
   |     ----------------------- help: other declaration is here
LL |
...
LL | contract D is C {
LL |     function f(uint8 a) public {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |

error[9914]: function overload clash during conversion to external types for arguments
  --> ROOT/tests/ui/typeck/external_type_clashes.sol:LL:CC
   |
LL |     function c(address) public pure {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
LL |
LL |     
LL |     function c(address payable) public pure {}
   |     ------------------------------------------ help: other declaration is here
   |

error[9914]: function overload clash during conversion to external types for arguments
  --> ROOT/tests/ui/typeck/external_type_clashes.sol:LL:CC
   |
LL |     function f(address) external pure {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
LL |
LL |     
LL |     function f(address payable) external pure {}
   |     -------------------------------------------- help: other declaration is here
   |

error[9914]: function overload clash during conversion to external types for arguments
  --> ROOT/tests/ui/typeck/external_type_clashes.sol:LL:CC
   |
LL |     function f(MyAddress a) external {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
LL |
LL |
LL |     function f(address a) external {}
   |     --------------------------------- help: other declaration is here
   |

error[9914]: function overload clash during conversion to external types for arguments
  --> ROOT/tests/ui/typeck/external_type_clashes.sol:LL:CC
   |
LL |     function g(MyAddress a) external {}
   |     ----------------------------------- help: other declaration is here
LL |
...
LL | contract H is G {
LL |     function g(I a) external {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |

error: aborting due to 6 previous errors

